shimmingtoolbox/unwrap/prelude.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+""" Description
+
+"""
+
+
+import numpy as np
+import os
+import nibabel as nib
+
+
+def prelude(complexArray, affine, mask=np.array([-1]), path2UnwrappedPhase="./unwrappedPhase.nii", isUnwrappingIn2D=True, isSavingNiftis=False):
+    """
+
+    Args:
+        complexArray:
+        affine:
+        mask:
+        path2UnwrappedPhase:
+        isUnwrappingIn2D:
+        isSavingNiftis:
+
+    Returns:
+
+    """
+    # Get absolute path
+    absPath = os.path.abspath(path2UnwrappedPhase)
+    dataSaveDirectory = os.path.dirname(absPath)
+
+    # Save complex image
+    # TODO: Save voxelSize when saving nifti (Maybe affine does it) (if it does, might be easier to just pass entire nibabel object)
+    phaseNii = nib.Nifti1Image(np.angle(complexArray[:-1, :-1, 0, 0]), affine)
+    magNii = nib.Nifti1Image(np.abs(complexArray[:-1, :-1, 0, 0]), affine)
+    nib.save(phaseNii, os.path.join(dataSaveDirectory, "rawPhase.nii"))
+    nib.save(magNii, os.path.join(dataSaveDirectory, "mag.nii"))
+
+    # Fill options
+    options = " "
+
+    if isUnwrappingIn2D:
+        options = options + "-s "
+
+    # Create mask with all ones if mask is not provided
+    if not mask.all() == -1:
+        assert mask.shape == complexArray.shape, "Mask must be the same shape as the array"
+        maskNii = nib.Nifti1Image(mask, affine)
+
+        options = options + "-m "
+        options = options + os.path.join(dataSaveDirectory, "mask.nii")
+        nib.save(maskNii, os.path.join(dataSaveDirectory, "mask.nii"))
+
+    # Unwrap
+    unwrapCommand = "prelude -p " + os.path.join(dataSaveDirectory, "rawPhase") + " -a " + \
```

Comment:
For this type of concatenation, I prefer to use the .format() function. It helps to see what are the types that you need to put in your string (like %d for integer or %f for float) and control how it is presented to make sure that no new version of Python can implicitly change how your string is formatted.